src/sentry/dynamic_sampling/latest_release_booster.py

```diff
@@ -1,234 +1,382 @@
 import re
 from dataclasses import dataclass, field
 from datetime import datetime
-from typing import Any, Dict, List, Optional, Tuple
+from typing import Any, Callable, Dict, List, Optional, Tuple
 
 from django.conf import settings
 from pytz import UTC
 
 from sentry.dynamic_sampling.latest_release_ttas import Platform
-from sentry.dynamic_sampling.utils import BOOSTED_RELEASES_LIMIT
 from sentry.models import Project, Release
 from sentry.utils import redis
 
-BOOSTED_RELEASE_TIMEOUT = 60 * 60
-ONE_DAY_TIMEOUT_MS = 60 * 60 * 24 * 1000
 ENVIRONMENT_SEPARATOR = ":e:"
 BOOSTED_RELEASE_CACHE_KEY_REGEX = re.compile(
     r"^ds::r:(?P<release_id>\d+)(:e:(?P<environment>.+))?$"
 )
 
 
-class TooManyBoostedReleasesException(Exception):
-    pass
+def get_redis_client_for_ds() -> Any:
+    cluster_key = getattr(settings, "SENTRY_DYNAMIC_SAMPLING_RULES_REDIS_CLUSTER", "default")
+    return redis.redis_clusters.get(cluster_key)
+
+
+def _get_environment_cache_key(environment: Optional[str]) -> str:
+    return f"{ENVIRONMENT_SEPARATOR}{environment}" if environment else ""
+
+
+def _get_project_platform(project_id: int) -> Platform:
+    try:
+        return Platform(Project.objects.get(id=project_id).platform)
+    except Project.DoesNotExist:
+        # If we don't find the project of this release we just default to having no platform name in the
+        # BoostedRelease.
+        return Platform()
+
+
+def _is_active(current_timestamp: float, expiration_timestamp: float) -> bool:
+    return current_timestamp <= expiration_timestamp
```

Comment:
I created this because I wanted to share the logic between the `is_active` method of boosted release and the inline comparison while adding data. It is not technically needed but it was just to centralize the logic because this is quite delicate.

Reply:
agree with @jjbayer let's remove it. You can add comment where you are using it
to say check if active etc.